core/src/main/java/hudson/model/ParameterValue.java

```diff
@@ -266,11 +264,9 @@ public boolean equals(Object obj) {
         if (getClass() != obj.getClass())
             return false;
         ParameterValue other = (ParameterValue) obj;
-        if (name == null) {
-            if (other.name != null)
-                return false;
-        } else if (!name.equals(other.name))
+        if (!Objects.equals(name, other.name)) {
             return false;
+        }
         return true;
```

Comment:
Can further be simplified to
```suggestion
        return Objects.equals(name, other.name);
```
but this seems easier to debug, so 🤷